standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/cache/CachedStore.java

```diff
@@ -900,13 +898,6 @@ private void updateTablePartitionColStats(RawStore rawStore, String catName, Str
               rawStore.getPartitionColumnStatistics(catName, dbName, tblName, partNames, colNames, CacheUtils.HIVE_ENGINE);
           Deadline.stopTimer();
           sharedCache.refreshPartitionColStatsInCache(catName, dbName, tblName, partitionColStats);
-          Deadline.startTimer("getPartitionsByNames");
-          List<Partition> parts = rawStore.getPartitionsByNames(catName, dbName, tblName, partNames);
-          Deadline.stopTimer();
-          // Also save partitions for consistency as they have the stats state.
-          for (Partition part : parts) {
-            sharedCache.alterPartitionInCache(catName, dbName, tblName, part.getValues(), part);
```

Comment:
Same concern here as previous. How do we fix this issue by not updating the cache at all ?

Reply:
The explanation is hidden in the commit message :)

Using alterPartitionInCache is problematic:
1. It empties all relevant stats about partitions even those updated just now.
2. It is not necessary since partitions were updated/refreshed just before calling the method updateTablePartitionColStats via [updateTablePartitions](https://github.com/apache/hive/blob/31ee14644bf6105360d6266baa8c6c8060d38ea3/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/cache/CachedStore.java#L870).
3. It is not meant to be used for syncing the cache but rather handle updates in partitions.

Reply:
Ok, thanks for the explanation. Would you need below lines, since you no longer use List<Partition>  ?

          Deadline.startTimer("getPartitionsByNames");
          List<Partition> parts = rawStore.getPartitionsByNames(catName, dbName, tblName, partNames);

Reply:
They are already removed aren't they? Maybe, I didn't understand what you mean.

Reply:
Ok, in review mode, only last two lines appear. I can see that they are removed in "Files changed" tab.

Reply:
Indeed some times its confusing :)